frontend/stores/watch-list.js

```diff
@@ -16,8 +16,28 @@ class WatchListStore {
   @action
   async fetch(userId) {
     try {
-      const res = await this.axios.get('/api/watch-manager/users/' + userId)
-      if (res.status >= 200 && res.status < 300) this.watches = res.data
+      const response = await axios.get('/api/watch-manager/users/' + userId)
+      if (response.status >= 200 && response.status < 300) {
+        const templates = await this.getTemplates()
+        const watchList = await response.data.map((watch) => {
+          const template = templates.find((value) => value._id === watch.templateID)
+          return { ...watch, templateName: template ? template.name : null }
+        })
+        this.watches = watchList
+      }
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  async getTemplates() {
+    try {
+      const response = await axios.get('/api/watch-manager/templates')
```

Comment:
```suggestion
      const response = await this.axios.get('/api/watch-manager/templates')
```
Xem lại hết nhé, ít nhất em cũng phải nhìn cả cái class xem nó có cái gì rồi chứ?
Và axios instance anh khởi tạo bên trên có `validateStatus = null`, tức là Promises resolve với mọi status code, từ 1xx-5xx